fcrepo-export/src/main/java/org/fcrepo/export/Exporter.java

```diff
@@ -0,0 +1,140 @@
+ */
+public class Exporter {
+    private static final Logger logger = getLogger(Exporter.class);
+    private Config config;
+    protected FcrepoClient client;
+    private URI binaryURI;
+    private URI containerURI;
+
+    /**
+     * Constructor that takes the Import/Export configuration
+     *
+     * @param config for export
+     */
+    public Exporter(final Config config) {
+        this.config = config;
+        this.client = FcrepoClient.client().build();
+        try {
+            binaryURI = new URI(RdfLexicon.NON_RDF_SOURCE.getURI());
+            containerURI = new URI(RdfLexicon.CONTAINER.getURI());
+        } catch (URISyntaxException ex) {
+            // no-op
+        }
+    }
+
+    /**
+     * This method does the export
+     */
+    public void run() {
+        System.out.println("Exporting!");
+        export(config.getResource());
+    }
+    private void export(final URI uri) {
+        try (FcrepoResponse response = client.head(uri).perform()) {
+            final List<URI> linkHeaders = response.getLinkHeaders("type");
+            if(linkHeaders.contains(binaryURI)) {
+                exportBinary(uri);
+            } else if (linkHeaders.contains(containerURI)) {
+                exportContainer(uri);
+            } else {
+                logger.error("Resource is neither an LDP Container nor an LDP NonRDFSource: {}", uri);
+            }
+        } catch (FcrepoOperationFailedException ex) {
+            logger.warn("Error retrieving content: {}", ex.toString());
+        } catch (IOException ex) {
+            logger.warn("Error writing content: {}", ex.toString());
+        }
+    }
+    private void exportBinary(final URI uri)
+            throws FcrepoOperationFailedException, IOException {
+        final File file = fileForBinary(uri);
+        if (file == null) {
+            logger.info("Skipping {}", uri);
+            return;
+        }
+
+        try (FcrepoResponse response = client.get(uri).perform()) {
+            logger.info("Exporting binary: {}", uri);
+            writeResponse(response, file);
+        }
+    }
+    private void exportContainer(final URI uri)
+            throws FcrepoOperationFailedException, IOException {
+        final File file = fileForContainer(uri);
+        if (file == null) {
+            logger.info("Skipping {}", uri);
+            return;
+        }
+
+        try (FcrepoResponse response = client.get(uri).accept(config.getRdfLanguage()).perform()) {
+            logger.info("Exporting description: {}", uri);
+            writeResponse(response, file);
+        } catch ( Exception ex ) { ex.printStackTrace(); }
+    }
+    void writeResponse(final FcrepoResponse response, final File file)
+            throws IOException {
+        Writer w = null;
+        try {
+            if (!file.getParentFile().exists()) {
+                file.getParentFile().mkdirs();
+            }
+            w = new FileWriter(file);
+            copy(response.getBody(), w, "UTF-8");
```

Comment:
This may be a later refinement, but consider using a reported size for the bitstream to select between `IOUtils::copy` and `::copyLarge`.